### PR TITLE
Fixed few loggers with mismatching placeholder-parameter count

### PR DESCRIPTION
**Pull request by kssimova**
See openhab/static-code-analysis#213
and openhab/static-code-analysis#237
Signed-off-by: Kristina S. Simova <kssimovaa@gmail.com>

### CODE REVIEW

**Comment by mherwege** [comment]
Sorry, missed the reference to the binding: please remove the logger argument in the Niko Home Control binding logger. It is not relevant in that warn message and could confuse the end user.

File: addons/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoConnection.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ public void send(final String cmd, final String value) {
         try {
             sendCommand(new EiscpMessage.MessageBuilder().command(cmd).value(value).build());
         } catch (Exception e) {
-            logger.error("Could not send command to device on {}: {}", ip + ":" + port, e);
+            logger.error("Could not send command to device on {}: {}", ip, port, e);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/OnkyoConnection.java]: I think the user meant` {}:{}: `

File: addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairEasyTouchHandler.java [hunk 2]
```diff
@@ -188,7 +188,7 @@ public void processPacketFrom(PentairPacket p) {
                 break;
             case 2:
                 if (p.getLength() != 29) {
-                    logger.debug("Expected length of 29: ", p);
+                    logger.debug("Expected length of 29: {}", p);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairEasyTouchHandler.java]: Does have a proper toString?

File: addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/BaseChannelHandler.java [hunk 3]
```diff
@@ -93,7 +93,7 @@ public final synchronized void removeAnySubscription(ConnectableDevice device) {
 
             @Override
             public void onError(ServiceCommandError error) {
-                logger.warn("{}: received error response: {}.", BaseChannelHandler.this.getClass().getName(), error);
+                logger.warn("{}: received error response: ", BaseChannelHandler.this.getClass().getName(), error);
```
[the remark this answers is not in the document]
Comment (kssimova) [reply addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/BaseChannelHandler.java]: Do you mean ServiceCommandError's toString method? It inherits class java.lang.Throwable's toString method, which returns the concatenation of: 
* the name of the class of this object
* ": " (a colon and a space)
* the result of invoking this object's getLocalizedMessage() method. If getLocalizedMessage returns null, then just the class name is returned.

File: addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/internal/handler/StopHandler.java [hunk 4]
```diff
@@ -190,7 +190,7 @@ private boolean fetchAndUpdateStopData() {
         try {
             uri = new URI(url);
         } catch (URISyntaxException e) {
-            logger.error("Unable to parse '%s' as a URI.", url);
+            logger.error("Unable to parse '%s' as a URI {}", url);
```
[the remark this answers is not in the document]
Comment (kssimova) [reply addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/internal/handler/StopHandler.java]: Yes, thank you!

File: addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairEasyTouchHandler.java (hunk 2 above)
Comment (kssimova) [reply addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairEasyTouchHandler.java]: The toString method of PentairPacket returns a hex String.

File: addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/handler/DSCAlarmBaseBridgeHandler.java [hunk 5]
```diff
@@ -647,7 +647,8 @@ public boolean sendCommand(DSCAlarmCode dscAlarmCode, String... dscAlarmData) {
                 }
 
                 if (password == null || password.length() < 1 || password.length() > 6) {
-                    logger.error("sendCommand(): Password is invalid, must be between 1 and 6 chars", password);
+                    logger.error("sendCommand(): Password is invalid, must be between 1 and 6 chars, password: {}",
```
[the remark this answers is not in the document]
Comment (kssimova) [reply addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/handler/DSCAlarmBaseBridgeHandler.java]: You are absolutely right! But as I am not the author of the class I am not sure what was the author's idea of logging the password. Should I open an issue and ask about it or should I just remove the password from logger? WDYT?

[the remark this answers is not in the document]
Comment (martinvw) [reply addons/binding/org.openhab.binding.dscalarm/src/main/java/org/openhab/binding/dscalarm/handler/DSCAlarmBaseBridgeHandler.java]: Just leave it out, please

**Pull request merged**